docs/pages/guides/prebuild.md

```diff
@@ -0,0 +1,168 @@
+
+In addition to generating the native folders, prebuild also makes the following modifications:
+
+- Generating a `metro.config.js` file.
+- Generating an `index.js` file.
+- Removing the `main` field in the `package.json`.
+- Modifying the `scripts` field in the `package.json`.
+- Modifying the `dependencies` field in the `package.json`.
+
+The convenience change to the `scripts` field is the only thing that alters how a user develops their app before/after prebuild, all other changes can be left as-is if desired.
+
+## Optionality
+
+Prebuilding is completely optional and does not disqualify a project from using other features or services offered by Expo. We created the designation _bare workflow_ to refer projects that do not use `npx expo prebuild`.
+
+Everything offered by Expo including [EAS][eas], Expo CLI, and the libraries in the Expo SDK are built to **fully support** the _bare workflow_ as this is a minimum requirement for supporting projects using `npx expo prebuild`.
+
+The main exception is the [Expo Go][expo-go] app which has partial support for the _bare workflow_, given the project in question only utilizes native features that are already available in the Expo Go app.
+
+We develop new [Native Modules][native-modules] without using `npx expo prebuild` as it's currently the fastest way to do so. However, we recommend users building apps with Expo attempt to use Prebuild as much as possible for the reasons listed in the [pitch](#pitch) section.
+
+## Pitch
+
+A single native project on its own is very complicated to maintain, scale, and grow. In a cross-platform app, you have multiple native projects that you must maintain and keep them up to date. We created the _optional_ Expo Prebuild system to streamline this process. Below are a few issues we've identified with native development and some corresponding reasons we believe Expo Prebuild solves these issues.
+
+### Sensible upgrades
+
+Building native code requires a basic familiarity with that native platform's default tooling leading to a fairly difficult learning curve. In cross-platform, this curve is multiplied by the amount of platforms you wish to develop for. Cross-platform tooling doesn't solve this issue if you need to drop down and implement many features individually, instead it often creates a harder system for developing apps.
+
+When you bootstrap a native app, it has a bunch of preset values and code that you don't need to understand in order to get started. Eventually you'll want to upgrade your native application and now you'll need to be acutely familiar with how all of the initial code works in order to safely upgrade it. This is extremely challenging and users will either upgrade their app incorrectly, missing crucial changes, or they'll bootstrap a new app and copy all of their source code into the new application.
```

Comment:
```suggestion
When you bootstrap a React Native app, there is a bunch of code and configuration that you don't need to understand in order to get started. But you now own it. Eventually, you will want to upgrade your native application and now you'll need to be familiar with how all of the initial code works in order to safely upgrade it. This is extremely challenging and users often either upgrade their app incorrectly, missing crucial changes, or they'll bootstrap a new app and copy all of their source code into the new application.
```

Reply:
The problem solved is not specific to React Native.